src/components/modals/AddEditModal.js

```diff
@@ -0,0 +1,195 @@
+  resize: none;
+`;
+
+const customStyles = {
+  control: (styles) => ({
+    ...styles,
+    backgroundColor: 'darkgrey',
+    border: 'none',
+    borderRadius: 0,
+    height: '57px',
+    boxShadow: 'none',
+    paddingLeft: '18px',
+  }),
+  input: (styles) => ({
+    ...styles,
+    color: 'white',
+    padding: 0,
+  }),
+  option: (styles) => ({
+    ...styles,
+    color: 'grey',
+  }),
+};
+
+const UseAddEditMovieModal = (flowType) => {
+  const [selectedMovieGenre, setSelectedMovieGenre] = useState({ name: 'comedy' });
+  const [releaseDate, setReleaseDate] = useState(new Date());
+
+  const CustomDatePickerInput = forwardRef((props, ref) => {
+    return (
+      <Datepicker onClick={props.onClick} ref={ref} role="button">
+        {props.value}
+      </Datepicker>
+    );
+  });
+
+  CustomDatePickerInput.displayName = 'CustomDatePickerInput';
+  const genres = [
+    { name: 'documentary' },
+    { name: 'comedy' },
+    { name: 'horror' },
+    { name: 'crime' },
+  ];
+  const movieGenres = useMemo(() => {
+    return genres.map((item) => {
+      return { value: item.name, label: item.name };
+    });
+  }, []);
+
+  const [showModal, hideModal] = useModal(() => (
+    <ReactModal isOpen style={modalStyles}>
+      <CloseIcon>
+        <p onClick={hideModal} role="button">
+          {' '}
+          X{' '}
+        </p>
+      </CloseIcon>
+      <h1>{flowType + ' Movie'}</h1>
+
+      <TopContainer>
+        <LeftContainer>
+          <FieldLabel>Title</FieldLabel>
+          <StyledInput placeholder="Movie title" />
+
+          <FieldLabel>Movie url</FieldLabel>
+          <StyledInput placeholder="https://" />
+
+          <FieldLabel>Genre</FieldLabel>
+          <Select
+            defaultValue={selectedMovieGenre}
+            onChange={(value) => setSelectedMovieGenre(value)}
+            options={movieGenres}
+            isMulti
+            components={{
+              IndicatorSeparator: () => null,
+            }}
+            placeholder="Select Genre"
+            styles={customStyles}
+          />
+        </LeftContainer>
+
+        <RightContainer>
+          <FieldLabel>Release date</FieldLabel>
+          <ReactDatePicker
+            dateFormat="dd/MM/yyyy"
+            onChange={(date) => setReleaseDate(date)}
+            selected={releaseDate}
+            customInput={<CustomDatePickerInput />}
+          />
+
+          <FieldLabel>Rating</FieldLabel>
+          <StyledInput placeholder="Movie rating" type="number" step="0.1" />
+
+          <FieldLabel>Runtime</FieldLabel>
+          <StyledInput placeholder="Minutes" type="number" />
+        </RightContainer>
+      </TopContainer>
+
+      <BottomContainer>
+        <FieldLabel>Overview</FieldLabel>
+        <StyledTextArea rows={10} placeholder={'Movie description'} />
+      </BottomContainer>
+
+      <ModalActions>
+        <PrimaryButtonOutlined>Reset</PrimaryButtonOutlined>
+        <PrimaryButtonFilled>Submit</PrimaryButtonFilled>
```

Comment:
Could you add event handlers for both 'Reset' and 'Submit' in both 'edit' and 'add' flows?